pkg/cmd/server/origin/auth.go

```diff
@@ -159,17 +275,32 @@ func (c *AuthConfig) NewOpenShiftOAuthClientConfig(client *oauthapi.Client) *osi
 		ClientSecret:             client.Secret,
 		ErrorsInStatusCode:       true,
 		SendClientSecretInParams: true,
-		AuthorizeUrl:             c.MasterPublicAddr + OpenShiftOAuthAPIPrefix + "/authorize",
-		TokenUrl:                 c.MasterAddr + OpenShiftOAuthAPIPrefix + "/token",
+		AuthorizeUrl:             OpenShiftOAuthAuthorizeURL(c.MasterPublicAddr),
+		TokenUrl:                 OpenShiftOAuthTokenURL(c.MasterPublicAddr),
 		Scope:                    "",
 	}
 	return config
 }
 
-func CreateOrUpdateDefaultOAuthClients(masterPublicAddr string, clientRegistry oauthclient.Registry) {
+func OpenShiftOAuthAuthorizeURL(masterAddr string) string {
+	return masterAddr + OpenShiftOAuthAPIPrefix + osinserver.AuthorizePath
```

Comment:
I know this is pre-existing from me, but I think we're supposed to use path.join.